src/core/queryClient.ts

```diff
@@ -123,6 +123,31 @@ export class QueryClient {
       .setData(updater, options)
   }
 
+  setQueriesData<TData>(
+    queryKey: QueryKey,
+    updater: Updater<TData | undefined, TData>,
+    options?: SetDataOptions
+  ): [QueryKey, TData][]
+
+  setQueriesData<TData>(
+    filters: QueryFilters,
+    updater: Updater<TData | undefined, TData>,
+    options?: SetDataOptions
+  ): [QueryKey, TData][]
+
+  setQueriesData<TData>(
+    queryKeyOrFilters: QueryKey | QueryFilters,
+    updater: Updater<TData | undefined, TData>,
+    options?: SetDataOptions
+  ): [QueryKey, TData][] {
+    return this.getQueryCache()
+      .findAll(queryKeyOrFilters)
+      .map(({ queryKey }) => [
+        queryKey,
+        this.setQueryData<TData>(queryKey, updater, options),
```

Comment:
@boschni I'm a bit worried that calling `setQueryData` multiple times in a row triggers multiple re-renders, and it does. see this example, where onButtonClick, I call `setQueryData` three times for three queries, and even though it's synchronous, it will trigger 3 re-renders:

https://codesandbox.io/s/tannerlinsleyreact-query-basic-forked-5c6rs?file=/src/index.js

I tried to wrap it in `notifyManager.batchCalls`, but it didn't change anything.

This is not much to do with this PR, but because I'm now calling setQueryData in a loop, it might show :)

Do you have any idea how to batch these calls maybe?

Reply:
couldn't find this out, but is also not so important I guess. if we get the api of that function right, we can always optimize it internally later :)

Reply:
Hi @TkDodo! The entire function body should be wrapped in `notifyManager.batch`, this should prevent redundant renders

Reply:
thank you 🙏 , seems like I messed up `batchCalls` and `batch` 😅 . fixed it here: 95bed800